react/src/views/Transfers/CreateShipment/CreateShipmentView.tsx

```diff
@@ -0,0 +1,250 @@
+      variables: {
+        baseId,
+      },
+    },
+  );
+
+  // Mutation after form submission
+  const [createShipmentMutation, createShipmentMutationState] = useMutation<
+    CreateShipmentMutation,
+    CreateShipmentMutationVariables
+  >(CREATE_SHIPMENT_MUTATION, {
+    update(cache, { data: returnedShipment }) {
+      cache.modify({
+        fields: {
+          shipments(existingShipments = []) {
+            const newShipmentRef = cache.writeFragment({
+              data: returnedShipment?.createShipment,
+              fragment: gql`
+                fragment NewShipment on Shipment {
+                  id
+                }
+              `,
+            });
+            return existingShipments.concat(newShipmentRef);
+          },
+        },
+      });
+    },
+  });
+
+  // Prep data for Form
+  const currentBase = allAcceptedTransferAgreements?.data?.base;
+  const currentOrganisationLabel = `${currentBase?.organisation?.name} - ${currentBase?.name}`;
+  const currentOrganisationId = globalPreferences.selectedOrganisationId?.toString();
+  const acceptedTransferAgreementsPartnerData =
+    allAcceptedTransferAgreements.data?.transferAgreements
+      ?.filter(
+        // Either we are the sourceOrg of the agreement or the targetOrg in combination with a bidirectional agreement
+        (agreement) =>
+          agreement.sourceOrganisation.id === currentOrganisationId ||
+          (agreement.targetOrganisation.id === currentOrganisationId &&
+            agreement.type === TransferAgreementType.Bidirectional),
+      )
+      .map((agreement) => {
+        // transform the agreement data to organisation base data
+        if (
+          agreement.targetOrganisation.id === currentOrganisationId &&
+          agreement.type === TransferAgreementType.Bidirectional
+        ) {
+          return {
+            id: agreement.sourceOrganisation.id,
+            name: agreement.sourceOrganisation.name,
+            bases: agreement.sourceBases,
+            agreementId: agreement.id,
+          } as IAcceptedTransferAgreementsPartnerData;
+        }
+        return {
+          id: agreement.targetOrganisation.id,
+          name: agreement.targetOrganisation.name,
+          bases: agreement.targetBases,
+          agreementId: agreement.id,
+        } as IAcceptedTransferAgreementsPartnerData;
+      });
+
+  const partnerOrganisationBaseData = acceptedTransferAgreementsPartnerData
+    ?.map(
+      (agreement) =>
+        // transform the agreement data to organisation base data
+        ({
+          id: agreement.id,
+          name: agreement.name,
+          bases: agreement.bases,
+        } as IOrganisationBaseData),
+    )
+    .reduce((accumulator, currentOrg) => {
+      // Merge options. If there are multiple transfer agreements this step is necessary
+      const existingOrganisation = accumulator.find((org) => org.id === currentOrg.id);
+      if (existingOrganisation) {
+        existingOrganisation.bases.push(
```

Comment:
I logged in as dev_headofops@boxcare.org and the CreateShipment page does not load. Console output says there's an error at this line
```
Uncaught TypeError: Cannot assign to read only property 'length' of object '[object Array]'
    at Array.push (<anonymous>)
```

Reply:
Can confirm that this is not reproducible with a clean DB container, I had more than one accepted agreement there before. So everything's ready for merging from my side, sorry about the wrong alarm